### DISCUSSION TITLE
I cloned the project instead of the package with python and git, how can I change the needed pytorch version.

### DISCUSSION THREAD

**Discussion by Javagaga** [lllyasviel/stable-diffusion-webui-forge#1520] [Q&A]
my local CUDA is 12.4 and pytorch 2.4, but if I ran web-user.bat it will install 2.3.1 pytorch and I don't want that, what should I do.

**Comment by AumGupta** [comment]
Please show the contents of your `web-user.bat` file.

**Comment by Haoming02** [comment]
> my local CUDA is 12.4 and pytorch 2.4

I assume you meant in the system wide global scope? **Forge** installs a copy of `PyTorch` in its own `venv`. There are two solutions:

- You can delete the `venv` folder, create a new one, then install `PyTorch 2.4` yourself first, before installing the rest of Forge later.
- Alternatively, modify [this line](https://github.com/lllyasviel/stable-diffusion-webui-forge/blob/main/modules/launch_utils.py#L365) to specify the desired versions before installing Forge. *(also modify the line above it to point to `cu124` instead of `cu121`)*
